### PR TITLE
feat: Add restack as a way to deploy Appsmith

### PR DESCRIPTION
**Pull request by kuberaspeaking**
This resolves [#17935](https://github.com/appsmithorg/appsmith/issues/17935).
Added Restack as a way to deploy Appsmith on Kubernetes on your own AWS infrastructure. 

## Docs added
- website/docs/getting-started/setup/installation-guides/restack.md
- website/static/img/Restack_Icon_100px.png
Screenshot 2022-11-11 at 17 52 40
Screenshot 2022-11-11 at 17 52 33

### CODE REVIEW

File: website/vercel.json [hunk 1]
```diff
@@ -77,6 +77,10 @@
       "source": "/setup/cloudjiffy",
```
Comment (jnikhila) [inline website/vercel.json:77]: Thanks @kuberaspeaking - for raising this PR. 

- The `vercel.json` changes are not needed. Please revert to the original.
- Could you please update the forked repo with the latest and then update the PR?

**Review by sharat87 — CHANGES_REQUESTED** [review]
Thank you for you contribution. I see that we are adding a dependency, `package.json`, but it doesn't look like we are using it anywhere. Can you revert the changes to both `package.json` and `package-lock.json` files please?

**Comment by kuberaspeaking** [comment]
@sharat87, those changes emerged when I updated my fork with the latest version of the repo. I haven't personally made any changes to those files. Reverting means going back to the non updated fork. Please advice on how to handle this.

File: website/docs/getting-started/setup/installation-guides/restack.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+You can deploy Appsmith on the Restack platform with just one click.
+
+The easiest way to do this is
+
+1. [Login to Restack](https://restack.io/login) with your GitHub or email.
+2. Connect your AWS account to create a cluster. 
+3. Search for Appsmith in the restack console and click on add application. 
+
+Here’s a [detailed guide](https://www.restack.io/docs/deploy-appsmith-on-restack#overview) provided in Restack’s docs.
```
Comment (jnikhila) [inline website/docs/getting-started/setup/installation-guides/restack.md:18]: @kuberaspeaking - the link (https://www.restack.io/docs/deploy-appsmith-on-restack#overview) gives a 404. Is this expected?

Comment (kuberaspeaking) [reply website/docs/getting-started/setup/installation-guides/restack.md:18]: The page slug was recently changed. Updating it in the docs.

**Comment by sharat87** [comment]
@kuberaspeaking, thanks for confirming. Looks like the diff is corrupted in this PR. Unfortunately, we cannot merge this while it's showing a diff of _deleting the package_ files.

We've gone ahead and raised a separate PR at https://github.com/appsmithorg/appsmith-docs/pull/577 for this and will be taking this forward from there.

Hope you understand. Thank you.

**Comment by kuberaspeaking** [comment]
Works for me.

**Pull request closed without merging**